### PR TITLE
Service contracts MUST NOT rely on the execution context

### PR DESCRIPTION
**Pull request by akaplya**
Service contracts MUST NOT rely on the execution context

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/devdocs?pullRequest=3870) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/devdocs?pullRequest=3870) before we can accept your contribution.<br/><hr/>**Anton Kaplia** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).

**Comment by magento-engcom-team** [comment]
@akaplya thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by magento-cicd2** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.2/coding-standards/technical-guidelines.md [hunk 1]
```diff
@@ -610,6 +610,8 @@ class View extends Template
 
 6.4.4.12. Any customizations to the domain/business logic MUST be executed on the Service Contracts layer, and so MUST be declared in the `global` area of configuration.
 
+6.4.4.13. Service contracts MUST NOT rely on the execution context(application area). Services implementation MUST NOT know about the application state.
```
Comment (dshevtsov) [inline guides/v2.2/coding-standards/technical-guidelines.md]:
```suggestion
6.4.4.13. Service contracts MUST NOT rely on the execution context (application area). Services implementation MUST NOT know about the application state.
```

Comment (akaplya) [reply guides/v2.2/coding-standards/technical-guidelines.md]: done

File: guides/v2.2/coding-standards/technical-guidelines.md [hunk 2]
```diff
@@ -610,6 +610,8 @@ class View extends Template
 
 6.4.4.12. Any customizations to the domain/business logic MUST be executed on the Service Contracts layer, and so MUST be declared in the `global` area of configuration.
 
+6.4.4.13. Service contracts MUST NOT rely on the execution context (application area). Services implementation MUST NOT know about the application state.
```
Comment (dshevtsov) [inline guides/v2.2/coding-standards/technical-guidelines.md]:
```suggestion
6.4.4.13. A service contract MUST NOT rely on execution context (application area). The service implementation MUST NOT depend on the application state.
```

Comment (melnikovi) [inline guides/v2.2/coding-standards/technical-guidelines.md]: Should it be SHOULD or can it be applied in all cases today?

Comment (akaplya) [reply guides/v2.2/coding-standards/technical-guidelines.md]: for all cases, for all service contracts

**Comment by dshevtsov** [comment]
running tests

**Comment by Vinai** [comment]
It would be helpful to add a reason for the rule to the document, or link to a resource explaining the rationale behind it.

**Pull request merged**